test/fixtures/TypeScript/main

```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/env node
+#!/usr/bin/env deno
```

Comment:
This file is not valid in Deno. (mostly due to imports)
1. Deno is not Node so you can not import packages like 'fs'
2. The file extension is important (`./x.ts` is valid but `./x` is not)
3. Deno does not use NPM, so `import * as tsickle from 'tsickle';`  is not right
etc...

Reply:
@pchaigno Could we get away with using a *"Hello, world"* type script for the TypeScript/JavaScript hashbang fixtures? I mean, the content of the script is kind of irrelevant after all...

Reply:
Sure, but if it's not too difficult to find a sample Deno file, it's even better. It'd help train the classifier to recognize the few differences of Deno.

Reply:
Ouch. Well, when I substitute the contents of `test/fixtures/TypeScript/main` for [this TypeScript program](https://github.com/denoland/deno_std/blob/master/examples/gist.ts), the tests fail, complaining that it identified as JavaScript. 😞

Do you think a heuristic is in order? Accurately differentiating TypeScript from JavaScript is impossible without a LOT of hairy regexes, but matching `from "[^"]+\.ts"` should work, since Node doesn't grok that extension.

Reply:
I've pushed my update. Have a look at the test feedback, I'm not really sure what's going on... 😢

Reply:
>  I'm not really sure what's going on... 😢

You do 😉 ...

> Accurately differentiating TypeScript from JavaScript is impossible without a LOT of hairy regexes

You've added `deno` to both JS and TS so things will fall through to heuristics and then the classifier and as there isn't a heuristic we're falling through to the classifier which due to the similarities of JS and TS has guessed this is JS not TS:

```console
$ LINGUIST_DEBUG=1 bundle exec bin/github-linguist test/fixtures/TypeScript/main
main: 66 lines (55 sloc)
  type:      Text
  mime type: text/plain
JavaScript =  -1455.108 +  -4.009 =  -1459.117
TypeScript =  -1591.191 +  -5.727 =  -1596.918
  language:  JavaScript
  appears to be a vendored file
$
```

Reply:
*Hold up.* I thought the point of these fixtures was to assert that the interpreter directives *were* recognisable?

Not gonna lie, the tests are completely and utterly opaque to me. 😢 Really not familiar with unit-testing in Ruby, but [this is what I'm used to reading](https://travis-ci.org/file-icons/atom/jobs/524356408#L772).

> so things will fall through to heuristics and then the classifier and as there isn't a heuristic we're falling through to the classifier which due to the similarities of JS and TS has guessed this is JS not TS:

Alrighty, sooo... the only sane solution is to remove Deno as a JavaScript interpreter. Which is okay, really, because TypeScript interpreters can compile JS, just not the other way around. 😁

Reply:
It's my fault for thinking too much about file extensions being tested before heuristics... of course, hashbangs are most relevant in files that don't *have* any extensions.

I'm an idiot, sorry. Deno's been pushed back into TypeScript-land. 👍 Hopefully the specs will pass now...